### PR TITLE
zh: add using br to backup to gcs and restore from gcs document

### PR DESCRIPTION
**Pull request by shuijing198799**
### What is changed, added or deleted? (Required)
Add using br to backup to gcs and restore from gcs document.

### Which TiDB Operator version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v1.1 (TiDB Operator 1.1 versions)
- [ ] v1.0 (TiDB Operator 1.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: 
- Other reference link(s):

### CODE REVIEW

File: zh/backup-to-gcs-br.md [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+  # backupType: full
+  from:
+    host: <tidb-host-ip>
+    port: <tidb-port>
+    user: <tidb-user>
+    secretName: backup-demo1-tidb-secret
+  br:
+    cluster: demo1
+    clusterNamespace: test1
+    # enableTLSClient: false
```
Comment (DanielZhangQD) [inline zh/backup-to-gcs-br.md:74]: It's removed

File: zh/backup-to-gcs-br.md [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+    projectId: <your-project-id>
+    secretName: gcs-secret
+    bucket: <my-bucket>
+    prefix: <my-folder>
+    # location: us-east1
+    # storageClass: STANDARD_IA
+    # objectAcl: private
+```
+
+以上示例中 `spec.br` 中的一些参数 `enableTLSClient`、`logLevel`、`statusAddr`、`concurrency`、`rateLimit`、`checksum`、`sendCredToTikv` 项均可省略。
```
Comment (DanielZhangQD) [inline zh/backup-to-gcs-br.md:91]: `enableTLSClient` is removed.

File: zh/backup-to-gcs-br.md [hunk 3]
```diff
@@ -0,0 +1,227 @@
+---
```
Comment (DanielZhangQD) [inline zh/backup-to-gcs-br.md:1]: The gcs docs for lightning+mydumper have updated, please update the docs based on the latest docs.

File: zh/backup-to-gcs-br.md (hunk 1 above)
Comment (shuijing198799) [reply zh/backup-to-gcs-br.md]: done

File: zh/backup-to-gcs-br.md (hunk 2 above)
Comment (shuijing198799) [reply zh/backup-to-gcs-br.md]: done

File: zh/backup-to-gcs-br.md (hunk 3 above)
Comment (shuijing198799) [reply zh/backup-to-gcs-br.md]: done

File: zh/backup-to-gcs-using-br.md [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+
+更多 `Backup` CR 字段的详细解释：
+
+* `.spec.metadata.namespace`：`Backup` CR 所在的 namespace。
+* `.spec.from.host`：待备份 TiDB 集群的访问地址。
+* `.spec.from.port`：待备份 TiDB 集群的访问端口。
+* `.spec.from.user`：待备份 TiDB 集群的访问用户。
+* `.spec.gcs.bucket`：存储数据的 bucket 名字。
+* `.spec.gcs.prefix`：这个字段可以省略，如果设置了这个字段，则会使用这个字段来拼接在远端存储的存储路径 `s3://${.spec.gcs.bucket}/${.spec.gcs.prefix}/backupName`。
+* `.spec.from.tidbSecretName`：待备份 TiDB 集群所需凭证的 secret。
```
Comment (DanielZhangQD) [inline zh/backup-to-gcs-using-br.md:145]:
```suggestion
* `.spec.from.tidbSecretName`：待备份 TiDB 集群 `.spec.from.user` 用户的密码所对应的 secret。
```

File: zh/backup-to-gcs-using-br.md [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+
+    如果 TiDB 集群开启了 [TLS](enable-tls-between-components.md)，但是不想使用[文档](enable-tls-between-components.md)中创建的 `${cluster_name}-cluster-client-secret` 进行备份，可以通过这个参数为备份指定一个 Secret，可以通过如下命令生成：
```
Comment (DanielZhangQD) [inline zh/backup-to-gcs-using-br.md:147]: 哪个参数？

File: zh/restore-from-gcs-using-br.md [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+
+以上示例将存储在 GCS 上指定路径 `spec.gcs.bucket` 存储桶中 `spec.gcs.prefix`文件夹下的备份数据恢复到 TiDB 集群 `spec.to.host`。关于 GCS 的配置项可以参考 [backup-gcs.yaml](backup-to-gcs-using-br.md#备份数据到-gcs) 中的配置。
+
+更多 `Restore` CR 字段的详细解释如下：
+
+* `.spec.metadata.namespace`： `Restore` CR 所在的 namespace。
+* `.spec.to.host`：待恢复 TiDB 集群的访问地址。
+* `.spec.to.port`：待恢复 TiDB 集群访问的端口。
+* `.spec.to.user`：待恢复 TiDB 集群的访问用户。
+* `.spec.to.tidbSecretName`：待恢复 TiDB 集群所需凭证的 secret。
```
Comment (DanielZhangQD) [inline zh/restore-from-gcs-using-br.md:102]:
```suggestion
* `.spec.to.tidbSecretName`：待恢复 TiDB 集群 `.spec.to.user` 用户的密码所对应的 secret。
```

File: zh/restore-from-gcs-using-br.md [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+
+    如果 TiDB 集群开启了 [TLS](enable-tls-between-components.md)，但是不想使用[文档](enable-tls-between-components.md)中创建的 `${cluster_name}-cluster-client-secret` 恢复备份，可以通过这个参数为恢复备份指定一个 Secret，可以通过如下命令生成：
```
Comment (DanielZhangQD) [inline zh/restore-from-gcs-using-br.md:104]: 哪个参数？

File: zh/backup-to-gcs-using-br.md [hunk 8]
```diff
@@ -0,0 +1,234 @@
...
+    projectId: ${project-id}
+    secretName: gcs-secret
+    bucket: ${bucket}
+    prefix: ${prefix}
+    # location: us-east1
+    # storageClass: STANDARD_IA
+    # objectAcl: private
+```
+
+以上示例中 `spec.br` 中的一些参数 `logLevel`、`statusAddr`、`concurrency`、`rateLimit`、`checksum`、`sendCredToTikv` 项均可省略。
```
Comment (DanielZhangQD) [inline zh/backup-to-gcs-using-br.md:90]: 没有介绍 cluster 和 cluster namespace。
Restore 文档也没有介绍。

Comment (shuijing198799) [reply zh/backup-to-gcs-using-br.md:90]: 加了 cluster 和 namespac 的介绍。restore 也加了说明

File: zh/restore-from-gcs-using-br.md (hunk 7 above)
Comment (shuijing198799) [reply zh/restore-from-gcs-using-br.md:105]: done

File: zh/restore-from-gcs-using-br.md (hunk 6 above)
Comment (shuijing198799) [reply zh/restore-from-gcs-using-br.md]: done

File: zh/backup-to-gcs-using-br.md (hunk 5 above)
Comment (shuijing198799) [reply zh/backup-to-gcs-using-br.md:152]: done

File: zh/backup-to-aws-s3-using-br.md [hunk 9]
```diff
@@ -322,7 +322,7 @@ Amazon S3 支持以下几种 `storageClass` 类型：
 * `.spec.from.port`：待备份 TiDB 集群的访问端口。
 * `.spec.from.user`：待备份 TiDB 集群的访问用户。
 * `.spec.from.tidbSecretName`：待备份 TiDB 集群 `.spec.from.user` 用户的密码所对应的 secret。
-* `.spec.from.tlsClient.tlsSecret`：指定备份使用的存储证书的 Secret。
+* `.spec.from.tlsClientSecretName`：指定备份使用的存储证书的 Secret。
```
Comment (DanielZhangQD) [inline zh/backup-to-aws-s3-using-br.md:325]: 这个改动不对

File: zh/restore-from-gcs-using-br.md [hunk 10]
```diff
@@ -0,0 +1,111 @@
...
+
+以上示例将存储在 GCS 上指定路径 `spec.gcs.bucket` 存储桶中 `spec.gcs.prefix`文件夹下的备份数据恢复到 TiDB 集群 `spec.to.host`。关于 BR、GCS 的配置项可以参考 [backup-gcs.yaml](backup-to-gcs-using-br.md#备份数据到-gcs) 中的配置。
+
+更多 `Restore` CR 字段的详细解释如下：
+
+* `.spec.metadata.namespace`： `Restore` CR 所在的 namespace。
+* `.spec.to.host`：待恢复 TiDB 集群的访问地址。
+* `.spec.to.port`：待恢复 TiDB 集群访问的端口。
+* `.spec.to.user`：待恢复 TiDB 集群的访问用户。
+* `.spec.from.tidbSecretName`：待备份 TiDB 集群 `.spec.from.user` 用户的密码所对应的 secret。
```
Comment (DanielZhangQD) [inline zh/restore-from-gcs-using-br.md:102]:
```suggestion
* `.spec.to.tidbSecretName`：待备份 TiDB 集群 `.spec.from.user` 用户的密码所对应的 secret。
```

File: zh/restore-from-gcs-using-br.md [hunk 11]
```diff
@@ -0,0 +1,111 @@
...
+* `.spec.from.tlsClientSecretName`：指定备份使用的存储证书的 Secret。
```
Comment (DanielZhangQD) [inline zh/restore-from-gcs-using-br.md:103]:
```suggestion
* `.spec.to.tlsClientSecretName`：指定备份使用的存储证书的 Secret。
```

Comment (shuijing198799) [reply zh/restore-from-gcs-using-br.md]: done

File: zh/restore-from-gcs-using-br.md (hunk 10 above)
Comment (shuijing198799) [reply zh/restore-from-gcs-using-br.md]: done

File: zh/backup-to-gcs-using-br.md [hunk 12]
```diff
@@ -0,0 +1,241 @@
+---
+title: 使用 BR 工具备份 TiDB 集群到 GCS
+summary: 介绍如何使用 BR 工具备份 TiDB 集群到 Google Cloud Storage (GCS)。
+category: how-to
+---
+
+# 备份 TiDB 集群到 GCS
+
+本文档详细描述了如何将 Kubernetes 上 TiDB 集群的数据备份到 [Google Cloud Storage (GCS)](https://cloud.google.com/storage/docs/) 上。本文档中的“备份”，均是指全量备份（Ad-hoc 全量备份和定时全量备份），底层通过使用 [`BR`](https://pingcap.com/docs-cn/v3.1/reference/tools/br/br) 获取集群的逻辑备份，然后再将备份数据上传到远端 GCS。
```
[the remark this answers is not in the document]
Comment (DanielZhangQD) [reply zh/backup-to-gcs-using-br.md:9]: Should change to `stable`.

File: zh/backup-to-gcs-using-br.md [hunk 13]
```diff
@@ -0,0 +1,241 @@
...
+
+    {{< copyable "shell-regular" >}}
+
+    ```shell
+    kubectl create secret generic backup-demo1-tidb-secret --from-literal=password=<password> --namespace=test1
+    ```
+
+### 备份数据到 GCS
+
+创建 `Backup` CR，并将数据备份到 GCS：
```
[the remark this answers is not in the document]
Comment (shuijing198799) [reply zh/backup-to-gcs-using-br.md:47]: there is only one step here， below is the example yaml。

File: zh/backup-to-gcs-using-br.md [hunk 14]
```diff
@@ -0,0 +1,241 @@
...
+
+`spec.br.statusAddr` 会为 BR 进程监听一个进程状态的 HTTP 端口，方便用户调试，如果不填，则默认不监听。
+
+`spec.br.concurrency` 的数量等于 备份/恢复 时每一个 TiKV 进程会使用的线程数。备份时默认为 4， 恢复时默认为 128。
+
+`spec.br.rateLimit` 代表了对流量进行限制。单位为 MB/S, 例如 设置为 `4` 代表了限速 4M/S, 默认不限速。
+
+`spec.br.checksum` 代表了是否在 备份/恢复 结束之后对文件进行验证。默认为 `true`。
+
+`spec.br.sendCredToTikv` BR 进程是否将自己的 AWS/GCP 权限传输给 TiKV 进程。默认为 `true`。
```
[the remark this answers is not in the document]
Comment (shuijing198799) [reply zh/backup-to-gcs-using-br.md]: done

File: zh/backup-to-gcs-using-br.md [hunk 15]
```diff
@@ -0,0 +1,241 @@
...
+    projectId: ${project-id}
+    secretName: gcs-secret
+    bucket: ${bucket}
+    prefix: ${prefix}
+    # location: us-east1
+    # storageClass: STANDARD_IA
+    # objectAcl: private
+```
+
+以上示例中 `spec.br` 中的一些参数 `logLevel`、`statusAddr`、`concurrency`、`rateLimit`、`checksum`、`sendCredToTikv` 项均可省略。
```
[the remark this answers is not in the document]
Comment (shuijing198799) [reply zh/backup-to-gcs-using-br.md]: done

File: zh/backup-to-gcs-using-br.md (hunk 12 above)
[the remark this answers is not in the document]
Comment (shuijing198799) [reply zh/backup-to-gcs-using-br.md]: done

**Comment by ran-huang** [comment]
@shuijing198799 I made some edits in [this commit](https://github.com/pingcap/docs-tidb-operator/pull/326/commits/6d0995a2ba1ee1a781c0b69cf4928f1ac76164c1). 
PTAL if they are okay. Thanks!

**Pull request merged**